### ISSUE TITLE
Documentation for javascript API

### ISSUE THREAD

**Issue by Marfusios** [riganti/dotvvm#234]
Hello,

is there somewhere a documentation for the javascript API of built-in controls and app life cycle? 

**Use case:**
I need to reload `GridView` every few seconds. 
Right now I ended up with something like this:

```
<dot:GridView ID="grid" class="table table-striped no-margin" DataSource="{value: Machines}">
    <Columns>
         ...
    </Columns>
</dot:GridView>

<dot:Button ID="reload_button" Click="{command: Reload()}" Text="Reload" hidden="hidden" />

<script>
    (function () {
        var timeout = 2000;
        var reloadFunction = function () {
            dotvvm.postbackScript('Reload()')('root', $('#reload_button')[0], [], '', false, "$root", null, null);
            setTimeout(reloadFunction, timeout);
        };

         setTimeout(reloadFunction, timeout);
    })();
</script>
```

`dot:Button` is just a workaround, b/c second parameter of method `postbackScript` must be a valid element (it's trying to access `tagName` property). 

It would be great to have something like:

```
dotvvm.reloadCurrentPage();
$('#grid').reload();
```

And definitely as parameters of `GridView`:

```
<dot:GridView AutoReloadingTimeout="2000" AutoReloadingMode="never|once|infinite">
```

**Comment by exyi** [comment]
The JS API is probably undocumented, since we don't consider it stable nor clean. I'd recommend you creating `Timer` control that will just invoke generated binding from JS. You'd probably still need to hack it a bit to invoke binding from JS code - see @tomasherceg's response at #179.

**Comment by tomasherceg** [comment]
Currently, the recommended way is to create a button, hide it using CSS, and click it in the timer. Please, do not call the DotVVM JS API directly as it might change in the future - we are still prototyping some things and we'll document it once we finish it.

```
<dot:Button style="display: none" ID="refresh-button" Click="{command: Refresh()}" />

<script>
window.setTimeout(function () { 
    $("#refresh-button").click(); 
}, 2000);
</script>
```

We plan to add the `Timer` control in the future, but it depends on one framework feature that is not finished yet.

**Comment by tomasherceg** [comment]
A better way to make a postback from JavaScript will hopefully be solved by #749 in DotVVM 3.0.

We've also added a docs page that describes how to interact with JS and client-side things: https://www.dotvvm.com/docs/tutorials/advanced-javascript-events-and-client-side-viewmodel/2.0

The proposed `AutoRefresh` properties might be useful on some controls, but I think that something like a `Timer` control would Be more useful as there may be more things that we'll need to update at the same time.

Feel free to reopen the issue, or open a new one.
